src/api/app/models/project.rb

```diff
@@ -192,14 +192,14 @@ def revoke_requests
     self.open_requests_with_project_as_source_or_target.each do |request|
       logger.debug "#{self.class} #{self.name} doing revoke_requests with #{@commit_opts.inspect}"
       # Don't alter the request that is the trigger of this revoke_requests run
-      next if request.id == @commit_opts[:request]
+      next if request == @commit_opts[:request]
```

Comment:
Yes sure rails, we're doing rails here right? :-)

Project.open_requests_with_project_as_source_or_target returns a 'ActiveRecord_Relation' of BsRequest instances.